src/tools/master_functions.cc

```diff
@@ -133,17 +153,126 @@ static int read_master_info(const char *name, master_info_t *info) {
+				        "SAUNAFS_STATUS_OK \n");
+				return SAUNAFS_ERROR_IO;
+			}
+			return status;
+		} else if (packet_version ==
+		           matocl::wholePathLookup::kResponsePacketVersion) {
+			matocl::wholePathLookup::deserialize(response, messageId, inode,
+			                                     attr);
+			return SAUNAFS_STATUS_OK;
+		} else {
+			fprintf(stderr,
+			        "SAU_MATOCL_WHOLE_PATH_LOOKUP: unknown version: %s \n",
+			        std::to_string(packet_version).c_str());
+			return SAUNAFS_ERROR_IO;
+		}
+	} catch (Exception &ex) {
+		fprintf(stderr, "SAU_MATOCL_WHOLE_PATH_LOOKUP: %s\n", ex.what());
+		return SAUNAFS_ERROR_IO;
+	}
+}
+
+size_t find_nth_occurrence(const std::string &str, char ch, int n) {
+	size_t pos = std::string::npos;
+
+	for (int i = 0; i < n; ++i) {
+		pos = str.find(ch, pos + 1);
+		if (pos == std::string::npos) {
+			return std::string::npos;
+		}
+	}
+
+	return pos;
+}
+
+void wsl_to_windows_path(char *rpath, size_t rpath_size) {
+	std::string wsl_path(rpath);
+	std::string windows_path;
+
+	// Find the position of the 5th occurrence of '\\'
+	size_t pos_system_folder = find_nth_occurrence(wsl_path, '\\', 4);
+	size_t pos_drive_letter = find_nth_occurrence(wsl_path, '\\', 5);
+
+	if (wsl_path[0] == '\\' && wsl_path[1] == '\\' &&
+	    pos_system_folder != std::string::npos) {
+		if (pos_drive_letter != std::string::npos &&
+		    wsl_path.substr(pos_system_folder, 5) == "\\mnt\\") {
+			char drive_letter = std::toupper(
+			    static_cast<unsigned char>(wsl_path[pos_drive_letter + 1]));
+
+			windows_path += drive_letter;
+			windows_path += ":\\";
+			windows_path += wsl_path.substr(pos_drive_letter + 3);
+
+			strncpy(rpath, windows_path.c_str(), rpath_size - 1);
+			rpath[windows_path.size()] = '\0';
+		} else {
+			windows_path = wsl_path.substr(pos_system_folder);
+			strncpy(rpath, windows_path.c_str(), rpath_size - 1);
+			rpath[windows_path.size()] = '\0';
+		}
+	}
+}
+#endif
+
+void get_next_path_iteration(std::string &path) {
+	size_t pos = std::min(path.find_first_of('\\'), path.find_first_of('/'));
+	if (pos != std::string::npos) {
+		path = path.substr(pos + 1);
+		pos = std::min(path.find_first_of('\\'), path.find_first_of('/'));
+		if (pos != std::string::npos) {
+			path = path.substr(pos);
+		} else {
+			path.clear();
+		}
+	} else {
+		path.clear();
+	}
+}
+
+int open_master_conn(const char *name, uint32_t *inode, mode_t *mode,
+                     [[maybe_unused]] bool needrwfs) {
 	char rpath[PATH_MAX + 1];
 	struct stat stb;
-	struct statvfs stvfsb;
+	[[maybe_unused]] struct statvfs stvfsb;
 	master_info_t master_info;
 
 	rpath[0] = 0;
-	if (realpath(name, rpath) == NULL) {
-		printf("%s: realpath error on (%s): %s\n", name, rpath, strerr(errno));
+
+	std::string name_to_use = std::string(name);
+#ifdef _WIN32
+	if (name[strlen(name) - 1] == '\\' || name[strlen(name) - 1] == '/') {
+		name_to_use = std::string(name).substr(0, strlen(name) - 1);
+	}
```

Comment:
Isn't this the same as (more or less):
```suggestion

	std::string name_to_use = std::string(name);
#ifdef _WIN32
	if (name_to_use.back() == '\\' || name_to_use.back() == '/') {
		name_to_use.pop_back();
	}
```
?

Reply:
Done.